eclair-core/src/main/scala/fr/acinq/eclair/channel/Helpers.scala

```diff
@@ -325,27 +325,41 @@ object Helpers {
     AnnouncementSignatures(channelParams.channelId, shortChannelId, localNodeSig, localBitcoinSig)
   }
 
-  /**
-   * This indicates whether our side of the channel is above the reserve requested by our counterparty. In other words,
-   * this tells if we can use the channel to make a payment.
+  /** Computes a maximum HTLC amount adapted to the current balance to reduce chances that other nodes will try sending payments that we can't relay.
    */
-  def aboveReserve(commitments: Commitments)(implicit log: LoggingAdapter): Boolean = {
-    commitments.active.forall(commitment => {
+  def maxHtlcAmount(nodeParams: NodeParams, commitments: Commitments): MilliSatoshi = {
+    if (!commitments.announceChannel) {
+      // The channel is private, let's not change the channel update needlessly.
+      return commitments.params.maxHtlcAmount
+    }
+    val availableToSend = commitments.active.map(commitment => {
```

Comment:
I actually looked for such a function but couldn't find it. I followed the code that checks if an HTLC can be relayed, expecting to find it there but `commitment.canSendAdd` does not use `commitment.availableBalanceForSend`, instead it duplicates most of the code. Is there a reason not to use `commitment.availableBalanceForSend` inside `commitment.canSendAdd`?

Reply:
They are mostly the same but not exactly the same, and in `canSendAdd` we need some of the intermediate values to report the error cases, that's why we actually duplicated most of this code.

It is very annoying that the protocol has such complexity around computing the actual available balance, but fortunately it will be greatly simplified in the future with `option_simplified_commitments` and 0-fee commit txs (once we have package relay).